### PR TITLE
Log in form

### PR DESCRIPTION
**Pull request by FrankieFabuloso**
- this is more than just the login form cause I forgot to branch.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -1,70 +1,38 @@
...
-      res.send(messages)
-    })
-  })
-})
+app.use( '/', render )
+app.use( '/chatroom', chatroom )
+app.use( '/user', user )
 
+//  add render middleware from routes/render
 server.listen(3000, function () {
```
Comment (jrobcodes) [inline app.js:36]: You probably want to make the port an environment variable, so that this can be deployed.

File: app.js [hunk 2]
```diff
@@ -74,20 +42,17 @@ io.on('connection', function(client) {
   console.log('client connetced...')
```
Comment (jrobcodes) [inline app.js:42]: Is this `console.log` intentional?

File: app.js [hunk 3]
```diff
@@ -74,20 +42,17 @@ io.on('connection', function(client) {
...
 
   client.on('join', function(data) {
```
Comment (jrobcodes) [inline app.js]: This is inconsistent with the ES6 you use elsewhere.

File: app.js [hunk 4]
```diff
@@ -74,20 +42,17 @@ io.on('connection', function(client) {
...
-    console.log(data)
     client.emit('messages', "Hello, from the sever...")
   })
 
   client.on('messages', function(data, req) {
```
Comment (jrobcodes) [inline app.js]: This is inconsistent with the ES6 you use elsewhere.

File: public/css/stylesheet.css [hunk 5]
```diff
@@ -0,0 +1,275 @@
+/* - - - - -  Landing Page CSS - - - - - */
```
Comment (jrobcodes) [inline public/css/stylesheet.css:1]: There's a lot going on in this file - consider splitting it up into multiple files for ease of maintenance.  You could create stylesheets for each of the sections you highlight, and either `link` them in the appropriate order, or (better) use tooling to combine and minify them.

File: public/scripts/home.js [hunk 6]
```diff
@@ -1,74 +1,89 @@
-const updateChatroomName = (chatroom_name) => {
-  $("span.current_convorsation").text(chatroom_name)
-  $(".converstaion_content").effect('bounce', 'slow')
+const updateChatroomName = (chatroomName) => {
```
Comment (jrobcodes) [inline public/scripts/home.js]: Style comment: you need not use parentheses on single parameter functions.

File: routes/chatroom/index.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+const express = require('express')
+const router = express.Router()
+const {Chatroom} = require('../../database/db.js')
+
+router.get('/', function( req, res) {
+  const { chatroom_name } = req.query
+
+  Chatroom.getChatroomIDByName(chatroom_name)
+  .then(result => Chatroom.getAllMessages(result.id) )
```
Comment (jrobcodes) [inline routes/chatroom/index.js]: indent for consistency

File: routes/render/index.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+const router = express.Router()
+const {User, Chatroom} = require('../../database/db.js')
+
+router.get('/', function (req, res) {
+  res.render('landing', {title: 'Messager'})
+})
+
+router.get('/home', function (req, res) {
+  const { user_id } = req.session
+    console.log('session:', req.session)
```
Comment (jrobcodes) [inline routes/render/index.js]: remove `console.log`

File: routes/user/index.js [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+} )
+
+router.post('/login', function( req, res, next ) {
+  const { email, username, password } = req.body
+
+  User.findByUsername( username )
+    .then( foundUser => {
+      if( checkValidLogin(foundUser.salt, foundUser.password, password ) ){
+        req.session.user_id = foundUser.id
```
Comment (jrobcodes) [inline routes/user/index.js]: You could simplify this to:
```js
req.session = {
  user_id: foundUser.id,
  username: foundUser.username,
  password: foundUser.password
}
```
Also, why are you storing the password in the session object?

File: socket.js [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+
+    client.on('join', function(data) {
+      client.emit('messages', "Hello, from the sever...")
+    })
+
+    client.on('messages', function(data, req) {
+      const { currentChatroom, newMessage, user_id } = data
+
+      Chatroom.getChatroomIDByName(currentChatroom)
+      .then(chatroom =>
```
Comment (jrobcodes) [inline socket.js]: indentation off

File: app.js (hunk 1 above)
Comment (FrankieFabuloso) [reply app.js:36]: in a .env?

File: public/css/stylesheet.css (hunk 5 above)
Comment (FrankieFabuloso) [reply public/css/stylesheet.css:1]: I had that before and Jared said to put it all into one big file....?
Which is more common or why is it better?
I was also given the feedback to use webpack to minify/bundle everything. Just haven't incorporated the feedback.

File: routes/user/index.js (hunk 9 above)
Comment (FrankieFabuloso) [reply routes/user/index.js]: Ooops! I didn't realize I was doing that.

File: app.js (hunk 2 above)
Comment (FrankieFabuloso) [reply app.js:42]: yes!

**Pull request merged**